### PR TITLE
 PT-3256: Add cart comment component, cart store 

### PR DESCRIPTION
**Pull request by ksavosteev**
Description: 
Added cart comment component and cart store 
--

QA-test: PT-3258

Demo-test: 

Download artifact URL: ghcr.io/virtocommerce/demo-xapi-app:1.0.0-pr-19-pt-3256-24616b1e

### CODE REVIEW

File: src/app/modules/checkout/components/cart-comment/cart-comment.component.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<div>
+    <label for="comment" class="form-label mt-4">Comments</label>
+    <textarea class="form-control no-resize" 
+        id="comment" 
+        name="comment" 
+        rows="5" 
+        [ngModel]='cartCommentText'
+        (ngModelChange)='onModelChange($event)'
+        maxlength="1024">
```
Comment (Vectorfield4) [inline src/app/modules/checkout/components/cart-comment/cart-comment.component.html]:
```suggestion
        maxlength="1024"
        rows="5"
        [ngModel]='cartCommentText'
        (ngModelChange)='onModelChange($event)'>
```

File: src/app/store/cart/cart.effects.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import { of } from 'rxjs';
+import { Apollo } from 'apollo-angular';
+import { ApolloError } from '@apollo/client/errors';
+import { cart, cartVariables } from 'src/app/graphql/types/cart';
+import getCartQuery from '../../graphql/queries/get-cart.graphql';
+import updateCartCommentMutation from '../../graphql/mutations/update-cart-comment.graphql';
+import * as CartActions from './cart.actions';
+import { updateCartComment, updateCartCommentVariables } from 'src/app/graphql/types/updateCartComment';
+import { Router } from '@angular/router';
+import { Store } from '@ngrx/store';
```
Comment (Vectorfield4) [inline src/app/store/cart/cart.effects.ts]: Sort imports.

File: src/app/store/cart/cart.effects.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    cartName: 'default',
+    currencyCode: 'USD',
+    cultureName: 'en-US',
+  }
+
+  getCart$ = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(CartActions.getCart),
+      concatMap(() => this.apollo.watchQuery<cart>(
+        {
```
Comment (Vectorfield4) [inline src/app/store/cart/cart.effects.ts]:
```suggestion
      concatMap(() => this.apollo.watchQuery<cart>({
```

File: src/app/store/cart/cart.effects.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        ))
+    );
+  });
+
+  setCartUserId$ = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(CartActions.setCartUserId),
+      tap(result => {
+        localStorage.setItem('cartUserId', result.userId);
+        void this.router.navigate([
```
Comment (Vectorfield4) [inline src/app/store/cart/cart.effects.ts]: I think we need to move this code out of here.

**Review by Vectorfield4 — APPROVED** [review]

**Review by eMazeika — APPROVED** [review]

**Pull request merged**